### PR TITLE
Move to Neovim 0.7

### LINKED ISSUE #163: Neovim v0.7 Support
This issue is just a placeholder for ongoing development discussion of Neovim v0.7 and getting AstroVim ready for it's release.

First things that come to mind:

- [x] migrate to Lua based autocommands
- [x] migrate to new keymap Lua API
- [x] global status line (`laststatus = 3`)
- [x] Lua based highlights. Purely Lua default theme

### PR DESCRIPTION
**Pull request by mehalter**
This PR adds support for using the new Lua APIs (auto commands, keymaps, and highlights) and the new global status bar that are being added to Neovim v0.7. 

## TODO

- [x] Drop backwards compatibility with neovim 0.6
- [x] Update README
  - [x] Requirements 0.6 -> 0.7
  - [x] Instructions to install with neovim 0.6 via the tag (no longer receiving updates)
- [x] Neovim 0.7 release
- [ ] Make sure all major package managers have the v0.7 available on stable
  - [x] [Homebrew](https://formulae.brew.sh/formula/neovim#default)
  - [ ] [MacPorts](https://ports.macports.org/port/neovim/)
  - [ ] [Arch Repos](https://archlinux.org/packages/community/x86_64/neovim/)
  - [ ] [Neovim PPA](https://launchpad.net/~neovim-ppa/+archive/ubuntu/stable)
  - [x] [EPEL](https://src.fedoraproject.org/rpms/neovim)
  - [ ] [Flathub](https://flathub.org/apps/details/io.neovim.nvim)
- [x] Make a `nvim-0.6` tag for last commit with support for below neovim 0.7
- [ ] Merge this
- [x] Rebrand to AstroNvim
- [ ] Rename repository to AstroNvim
- [ ] Migrate Repository

Resolves #163

### CODE REVIEW

**Comment by mehalter** [comment]
Neovim v0.7.0 has been officially released as stable (https://github.com/neovim/neovim/releases/tag/v0.7.0)

We will continue to monitor the listed package repositories and wait until v0.7.0 is generally available before merging this and doing the rebranding. Hopefully it doesn't take too long and we can make this release in the coming days.

**Comment by mehalter** [comment]
Going to go ahead and mark this as ready for review instead of a draft PR. Still going to hold off on merging until package repos update with Neovim v0.7

**Review by kabinspace — APPROVED** [review]

**Pull request merged**
